### DISCUSSION TITLE
Add event with custom JS call ?

### DISCUSSION THREAD

**Discussion by yellowbean** [reflex-dev/reflex#3036] [Q&A] [answered]
Hi, guys

Thanks for your all kind support and great work ! 

I've encountered an issue, I've build a custom js library and would like to call functions inside of it. 

the function can be called in developer console after including:

    rx.script(src="/my-custom.js"),

The question is how to call custom function by a button click event ? pseudo code may look like this 

    rx.button("Highlight Diff", on_click=rx.script("myCustFunction()")),

**Comment by yellowbean** [comment]
The workaround :

      rx.script("""u('#hlightBtn').on('click', function(e) {
                          myCustFunction();
                  });"""),
      rx.button("Highlight Diff",id="hlightBtn"),

But it just adding too much noise

**Comment by picklelo** [comment, marked as answer]
@yellowbean Does your original code with `rx.call_script` not work? That's the recommended way of doing this.

**Reply by masenf** [reply]
```python
rx.button("Highlight Diff", on_click=rx.call_script("myCustFunction()")),
```

This should work

**Reply by yellowbean** [reply]
hmm. I didn't find this function in https://reflex.dev/docs/library/other/script/ . Thanks, let me try it out

**Reply by masenf** [reply]
it's described on this page https://reflex.dev/docs/api-reference/browser-javascript/

probably should link that from the script component doc
